android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java

```diff
@@ -53,11 +69,84 @@ public void present(@NonNull String document, @NonNull ReadableMap configuration
         }
     }
 
+    @ReactMethod
+    public void setPageIndex(final int pageIndex, final boolean animated) {
```

Comment:
Lines 74-75 introduce a possibility for race (since the `currentActivity` might be replaced between type check and cast). (unless this method runs on the main thread – but it looks like it doesn't)